src/borrow_tracker/mod.rs

```diff
@@ -274,20 +274,20 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         &mut self,
         kind: RetagKind,
         place: &PlaceTy<'tcx, Provenance>,
-        borrow_tracker_method: BorrowTrackerMethod,
     ) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
-        match borrow_tracker_method {
+        let method = this.machine.borrow_tracker.as_ref().unwrap().borrow().method();
```

Comment:
Why do you call `method()` here? You should have access to the private field, it's in the same module. (Same below a few more times. Is `fn method` even needed?)